tls/src/params.cc

```diff
@@ -216,7 +228,8 @@ void params::validate_cert(gnutls_session_t session) {
   if (!_ca.empty()) {
     int ret;
     uint32_t status;
-    ret = gnutls_certificate_verify_peers2(session, &status);
+    ret = gnutls_certificate_verify_peers3(session, _tls_hostname.c_str(),
+                                           &status);
```

Comment:
if tls_hostname is not provided by the user, it will be seen as an empty string and not as nullptr as it should be I think...